File: ios/Crew/Engine/MealTag.swift
// SPEC: Flow 4 time-smart tags · Decision Registry G10 boundaries. Twin of meal-tag.ts. WRITTEN — UNVERIFIED (needs Mac).

import Foundation

enum MealTag: String, CaseIterable, Codable {
    case breakfast, lunch, dinner, snack

    var emoji: String {
        switch self {
        case .breakfast: return "🍳"
        case .lunch: return "🥗"
        case .dinner: return "🍽"
        case .snack: return "🥤"
        }
    }

    static func tagFor(minuteOfDay: Int) -> MealTag {
        if minuteOfDay >= SpecConstants.mealTagBreakfastFromMinute && minuteOfDay < SpecConstants.mealTagLunchFromMinute { return .breakfast }
        if minuteOfDay >= SpecConstants.mealTagLunchFromMinute && minuteOfDay < SpecConstants.mealTagDinnerFromMinute { return .lunch }
        if minuteOfDay >= SpecConstants.mealTagDinnerFromMinute && minuteOfDay < SpecConstants.mealTagDinnerUntilMinute { return .dinner }
        return .snack
    }

    static func tagFor(date: Date, timeZone: TimeZone = .current) -> MealTag {
        var calendar = Calendar(identifier: .gregorian)
        calendar.timeZone = timeZone
        let components = calendar.dateComponents([.hour, .minute], from: date)
        return tagFor(minuteOfDay: (components.hour ?? 0) * TimeUnits.minutesPerHour + (components.minute ?? 0))
    }
}

File: web/src/lib/engine/meal-tag.ts
// SPEC: Flow 4 time-smart tags (7 AM → 🍳 · 12:30 → 🥗 · 7 PM → 🍽 · odd hours → 🥤) · Decision Registry G10 (breakfast
// 04:00–10:30 · lunch 10:30–15:30 · dinner 15:30–21:00 · snack otherwise), local minutes of the day. Twin: MealTag.swift.
import { SpecConstants } from "@/generated/spec-constants";

export type MealTag = "breakfast" | "lunch" | "dinner" | "snack";

export function mealTagFor(minuteOfDay: number): MealTag {
  if (minuteOfDay >= SpecConstants.mealTagBreakfastFromMinute && minuteOfDay < SpecConstants.mealTagLunchFromMinute) return "breakfast";
  if (minuteOfDay >= SpecConstants.mealTagLunchFromMinute && minuteOfDay < SpecConstants.mealTagDinnerFromMinute) return "lunch";
  if (minuteOfDay >= SpecConstants.mealTagDinnerFromMinute && minuteOfDay < SpecConstants.mealTagDinnerUntilMinute) return "dinner";
  return "snack";
}

export const mealTagEmoji: Record<MealTag, string> = { breakfast: "🍳", lunch: "🥗", dinner: "🍽", snack: "🥤" };
